### PR TITLE
Rename MULTIRUST_ env vars to RUSTUP_. Leave compatibility code.

### PR DESCRIPTION
**Pull request by brson**
Pretty self explanatory.

### CODE REVIEW

**Comment by alexcrichton** [comment]
r=me, but looks like tests are failing?

**Comment by brson** [comment]
I was getting crashes accessing a `String` inside `defer!` on both MSVC targets, so I replaced all uses with `scopeguard::guard`, converting the captures to explicit arguments. Perhaps scopeguard contains some unsoundness cc @bluss @alexcrichton.

File: tests/cli-self-update.rs [hunk 1]
```diff
@@ -38,8 +37,8 @@ pub fn setup(f: &Fn(&Config)) {
 
         // An windows these tests mess with the user's PATH. Save
         // and restore them here to keep from trashing things.
-        let ref saved_path = get_path();
```
Comment (bluss) [inline tests/cli-self-update.rs]: This may be totally baseless, but could it be related to this uncommon use of `ref` here?

**Comment by bluss** [comment]
Thank you for the ping! Scopeguard is rather simple and pure safe rust, so it would be a soundness bug in rustc.

source: https://github.com/bluss/scopeguard/blob/master/src/lib.rs

File: tests/cli-self-update.rs (hunk 1 above)
Comment (brson) [inline tests/cli-self-update.rs]: I believe I also converted it to a value binding and still got the crash, but can't be sure.

**Pull request merged**